pyflakes/test/test_is_literal.py

```diff
@@ -198,3 +198,25 @@ def test_chained_operators_is_str_end(self):
         if 4 < x is 'foo':
             pass
         """, IsLiteral)
+
+    def test_is_tuple_constant(self):
+        self.flakes('''\
+            x = 5
+            if x is ():
+                pass
+        ''', IsLiteral)
+
+    def test_is_tuple_constant_containing_constants(self):
+        self.flakes('''\
+            x = 5
+            if x is (1, '2', True, (1.5, ())):
+                pass
+        ''', IsLiteral)
+
+    def test_is_tuple_containing_variables_ok(self):
+        # a bit nonsensical, but does not trigger a SyntaxWarning
+        self.flakes('''\
+            x = 5
+            if x is (x,):
```

Comment:
weird i wonder why this doesn't trigger and if that is intentional on python's end

Reply:
this tuple isn't a constant literal 🤷‍♂

Reply:
Maybe the trigger should be "anything other than a name or named constant"? It doesn't really make sense for non-constant literals to pass where constant literals fail, they're even more obviously incorrect.

Reply:
the goal is to match the cpython warning